src/wired/error_boundry.tsx

```diff
@@ -1,32 +1,52 @@
 import Error, { ErrorProps } from 'next/error';
 import React, { Component, PropsWithChildren } from 'react';
+import { NextPageContext } from 'next';
 
 export type WiredErrorBoundaryProps = PropsWithChildren<{
-  ErrorComponent?: React.ComponentType<ErrorProps>;
+  ErrorComponent?: React.ComponentType<
+    ErrorProps & { error: NextPageContext['err'] }
+  >;
 }>;
 
 interface WiredErrorBoundaryState {
-  hasError: boolean;
+  error?: NextPageContext['err'];
 }
 
 export class WiredErrorBoundary extends Component<
   WiredErrorBoundaryProps,
   WiredErrorBoundaryState
 > {
-  static getDerivedStateFromError(): WiredErrorBoundaryState {
-    return { hasError: true };
+  static getDerivedStateFromError(
+    error: NextPageContext['err']
+  ): WiredErrorBoundaryState {
+    return { error };
   }
 
-  state = { hasError: false };
-
   render() {
     const ErrorComponent = this.props.ErrorComponent;
 
-    if (this.state.hasError) {
+    // Something happened, let consumers decide what to do with this type of error
+    let error: NextPageContext['err'];
+
+    // Because passing this.state.error won't work
+    if (this.state.error)
+      error = {
+        name: 'Error 500',
+        message: this.state.error ? this.state.error.toString() : 'Unknown',
+        stack: this.state.error?.stack
+          ? this.state.error.stack.toString()
+          : 'Unknown',
+        statusCode: 500,
+      };
+
+    // eslint-disable-next-line no-console
+    console.log('error_boundary:error', error);
+
+    if (error) {
       return ErrorComponent ? (
-        <ErrorComponent statusCode={500} />
+        <ErrorComponent statusCode={500} error={error} />
```

Comment:
Can `this.state.error` be passed in directly here rather than wrapping it with our own `name` and `statusCode`? It seems like the types should work out.

Reply:
@rrdelaney That was actually my first approach. I was passing it directly and then doing `JSON.stringify(props)` on my ErrorComponent. It wouldn't be passed. That's why I went for the approach that resembled the Error structure, but with a flat object

Reply:
I think `JSON.stringify` misses `Error`'s, so that may be why that was missing. If it works other than that I think it would be better to let users write their own serialization logic rather than decide how they're errors should be structured.

Reply:
Re-running some tests locally, btw, just to be sure. However, i'm sure I logged props as well and error would return empty. I believe it may be a React limitation. Confirming as we speak, but IIRC this is why I had this line:
https://github.com/RevereCRE/relay-nextjs/pull/49/files#diff-507d8a2147310e2c2fa3d244db6af67f5353bad363e439df3f1e03122b0e0959R31